transports/jms/src/main/java/org/mule/transport/jms/JmsMessageDispatcher.java

```diff
@@ -186,6 +186,8 @@ private MuleMessage dispatchMessage(MuleEvent event, boolean doSend) throws Exce
 
             if (useReplyToDestination && replyTo != null)
             {
+                final int timeout = endpoint.getResponseTimeout();
```

Comment:
There is no way to check if endpoint value is set (by user) or not, as it is an int with a default value.  This behaviour while different to Mule 3.1 and before is consistent with other connectors like HTTP.

The fact that timeout is now respected I don't think belongs in the migration guide, but the fact that you can no longer customize JMS request-response timeout via event.setTimeout() i'll add a note on.